python/ray/tune/tests/test_trial_scheduler_pbt.py

```diff
@@ -210,7 +217,7 @@ def tearDown(self):
 
     def synchSetup(self, synch, param=None):
         if param is None:
-            param = [10, 20, 30]
+            param = [10, 20, 40]
```

Comment:
Why do we need to change this?

Reply:
If I remember it correctly, this test needs to showcase async pbt v.s. sync pbt. So it was originally set up to be:
async:
trial1 result --> trial2 result --> trial1 result --> perturb --> trial whatever result
sync:
trial1 result --> trial2 result --> trial1 result --> .. --> trial3 result --> perturb

The sequence is important to showcase the difference between sync and async. Specifically, trial1 should proceed to `iter=2` strictly before trial3 does.

Having `30` is giving race condition to the above sequence. This is because
trial1 iter=1 (ts = 10)--> pause --> trial2 iter=1 (ts = 20) --> (I) trial2 paused and trial1 is running again --> (II) trial3 iter=1 (ts = 30) --> (III) trial1 iter=2 (ts = 10 + 20 + epsilon)

Because of this change, step (I) needs to happen in the next tune loop (because pg is not cached and readily to be used). This small epsilon manifests the racing condition, that was originally there in the set up (30 = 10 + 20)